### PR TITLE
Feature/map options #65

### LINKED ISSUE #65: Additional Attributes Example Error: navigationMode must be 'css-transforms' or 'classic'.
Only appears to happen when linking to that example after viewing other examples, does not appear to happen when you go directly to the example page (http://esri.github.io/angular-esri-map/#/examples/additional-attributes).

### PR DESCRIPTION
**Pull request by jwasil**
@tomwayson please take a look. (Hopefully) resolves #65. Please see commit comments for high-level changes.

### CODE REVIEW

**Comment by jwasil** [comment]
Additional to do:
- [ ]     Rename related docs and test pages to "additional-map-options" (Additional Map Options)
- [ ]     Update example docs page routes, contoller, view, etc. to accommodate renamed files
- [ ]     Add link from Additional Map Options docs page to the related test page
- [ ]     Update test page to use inline JSON for mapOptions, but update the example docs page to use $scope.mapOptions defined in controller
- [ ]     Read the map options above the check for web map id; createMap() method can also accept the mapOptions object
- [ ]     Create new test page or example docs page to load a map using a non-web Mercator projection as well as construct a Popup. Pass the extent and popup through the mapOptions. Do not supply center/zoom/basemap. Use extentChange to report changes to the extent on the page. Remove $attrs.extent convenience attribute, and add new code in the `esriMap` module for Extent and Popup construction.

**Comment by tomwayson** [comment]
Go @jwasil , it's your birthday :cake:!

New examples/test page look great. It occurs to me that one of the two should probably load a web map to demonstrate the capabilities of passing the options when loading a web map. You can either add that to your list above if you want to handle it in this PR, or create a new issue for the backlog.

File: docs/app/examples/additional-map-options.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                sliderOrientation: 'horizontal',
+                sliderPosition: 'top-right',
+                displayGraphicsOnPan: true,
+                fadeOnZoom: false,
+                logo: false,
+                wrapAround180: false
+            }
+        };
+
+        /*esriLoader.require(['esri/geometry/Extent', 'esri/SpatialReference'], function(Extent, SpatialReference) {
```
Comment (tomwayson) [inline docs/app/examples/additional-map-options.js]: should remove this comment block b/c anything in this file shows up the code snippet on the examples page:

image

Comment (jwasil) [inline docs/app/examples/additional-map-options.js]: @tomwayson thank you for reviewing what is in here so far! You're completely right about the commented out code showing up in the front-facing code snippet.  That's just sitting in there temporarily until I move it out to a new example/test page that loads in an extent and infoWindow from the mapOptions.

Comment (tomwayson) [inline docs/app/examples/additional-map-options.js]: Gotcha! I'll wait for all the checkboxes to be ticked before final review. Thanks!

**Comment by jwasil** [comment]
@tomwayson good point on showing a webmapId and mapOptions jiving together.  I've updated the to do list above.  Feel free to modify the to do list if anything else comes to mind. Thanks :hand:-5

**Comment by jwasil** [comment]
Hey @tomwayson feel free to start digging in! The new "Advanced Map Options" example page is up, which required changes in a few spots to the esriMap directive; the most recent commit attempts to compartmentalize those changes.  Looking forward to your thoughts on how the Extent and Popup are constructed, particularly the srcNodeRef param on the Popup. Thanks!

**Comment by tomwayson** [comment]
Great work @jwasil! I made a few minor changes.

I really appreciate you digging into the extent and info window options on top of everything else. I like the way you check if the extent needs to be created or not.

I'm actually still playing w/ a few different things like checking for scope properties like basemap, center, and zoom before checking for the web map id, etc. I just wanted to get this thing merged b/c we've gone so far beyond the original bug fix and we can handle any future changes under new issues if need be.

**Comment by jwasil** [comment]
Awesome, @tomwayson! Thanks for reviewing this, providing advice, and doing some additional work on this PR.

I just noticed the "Additional Map Options" test page is not linked to properly anymore from the "No Basemap" example docs page. I realize I set up the href through the relative path to what the gulp task exposed during dev.

If it doesn't step on something you're looking into, I can do some quick changes to link to the unserved git source as well as update file names and titles.

**Comment by jwasil** [comment]
^ Please disregard my confusion b/w additional map options and no basemap above. This was written before :coffee:! The only thing is how to handle the link to the additional map options test page.

**Comment by tomwayson** [comment]
Good catch. Please look into why that's not working - the link is right, I just don't know why the test file didn't get deployed.

**Comment by jwasil** [comment]
done and deployed to gh-pages as well  :shipit:

**Pull request merged**